Source/WebCore/loader/HTTPHeaderField.cpp

```diff
@@ -30,16 +30,31 @@
 
 namespace WebCore {
 
-std::optional<HTTPHeaderField> HTTPHeaderField::create(String&& unparsedName, String&& unparsedValue)
+HTTPHeaderField::HTTPHeaderField(String&& name, String&& value)
+{
+    StringView strippedName = StringView(name).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    StringView strippedValue = StringView(value).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    
+    ASSERT(RFC7230::isValidName(strippedName) && RFC7230::isValidValue(strippedValue));
```

Comment:
I don't think this assertion is valid when decoding from an untrusted source, even after stripping whitespace.